### PR TITLE
TSK-1079 Баг Отображение Неактивных пустых групп в разделе Активных задач

### PR DESCRIPTION
**Pull request by BykhovDenis**
# Contribution checklist

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues
https://front.hc.engineering/workbench/platform/tracker/TSK-1079
A list of closed updated issues

### CODE REVIEW

File: plugins/view-resources/src/components/list/ListCategories.svelte [hunk 1]
```diff
@@ -85,7 +85,11 @@
           const f = await getResource(categoryFunc.action)
           const res = hierarchy.clone(await f(_class, query, groupByKey, update, queryId, $statusStore))
           if (res !== undefined) {
-            categories = Array.from(new Set([...categories, ...res]))
+            const set = new Set(categories.map((p) => JSON.stringify(p)))
```
Comment (haiodo) [inline plugins/view-resources/src/components/list/ListCategories.svelte]: Why we need this? They already should be uniq.

File: plugins/view-resources/src/viewOptions.ts [hunk 2]
```diff
@@ -131,16 +131,18 @@ export async function showEmptyGroups (
...
       return it.ofAttribute === attr._id
     })
     if (query !== undefined) {
-      statusList = matchQuery<Status>(
-        statusList,
-        query as DocumentQuery<Status>,
-        _class,
-        hierarchy
-      ) as unknown as Array<WithLookup<Status>>
+      const { [key]: st, space } = query
```
Comment (haiodo) [inline plugins/view-resources/src/viewOptions.ts:138]:
```suggestion
      const { [key]: st, space, ...data } = query
```

File: plugins/view-resources/src/viewOptions.ts [hunk 3]
```diff
@@ -131,16 +131,18 @@ export async function showEmptyGroups (
...
+      const resQuery: DocumentQuery<Status> = {
+        space
```
Comment (haiodo) [inline plugins/view-resources/src/viewOptions.ts:140]:
```suggestion
        ...data,
        space
```

File: plugins/view-resources/src/components/list/ListCategories.svelte (hunk 1 above)
Comment (BykhovDenis) [reply plugins/view-resources/src/components/list/ListCategories.svelte]: This is necessary when the function returns a filtered value, but we have an object with a different category field value. It worked fine until we only had the number | string as category, now we have objects (status) and Set can't make it unique.

Comment (BykhovDenis) [reply plugins/view-resources/src/components/list/ListCategories.svelte]: Probably better use something like fastEqual, but I'm not sure

File: plugins/view-resources/src/viewOptions.ts (hunk 2 above)
Comment (BykhovDenis) [reply plugins/view-resources/src/viewOptions.ts:138]: We don't need it for status filter

File: plugins/view-resources/src/viewOptions.ts (hunk 3 above)
Comment (BykhovDenis) [reply plugins/view-resources/src/viewOptions.ts:140]: We don't need it for status filter

File: plugins/view-resources/src/components/list/ListCategories.svelte (hunk 1 above)
Comment (haiodo) [reply plugins/view-resources/src/components/list/ListCategories.svelte]: categories = await getCategories(client, _class, docs, groupByKey, $statusStore) should always return uniq values already.

File: plugins/view-resources/src/utils.ts [hunk 4]
```diff
@@ -648,6 +648,33 @@ export async function getCategories (
...
+      const id = item.name
+      if (!uniqueObjects.has(id)) {
+        uniqueObjects.set(id, item)
+      }
+    } else {
+      uniqueValues.add(item)
+    }
+  }
+
+  return [...uniqueValues, ...uniqueObjects.values()]
```
Comment (haiodo) [inline plugins/view-resources/src/utils.ts:675]: Order will be lost if we mix StatusValue and other values.

**Review by haiodo — APPROVED** [review]

**Pull request merged**